src/Momento.Sdk/Auth/EnvironmentTokenProvider.cs

```diff
@@ -0,0 +1,34 @@
+namespace Momento.Sdk.Auth;
+
+using System;
+using Momento.Sdk.Exceptions;
+using Momento.Sdk.Internal;
+
+public class EnvironmentTokenProvider : ICredentialProvider
+{
+    public string AuthToken { get; private set; }
+    public string ControlEndpoint {get; private set; }
+    public string CacheEndpoint { get; private set; }
+
+    /// <summary>
+    /// Reads and parses a JWT token stored as an environment variable.
+    /// <param name="name">Name of the environment variable that contains the JWT token.</param>
+    /// </summary>
+    public EnvironmentTokenProvider(string name)
```

Comment:
I am sorry, I know I said ship it on the previous pass. I wanted to check how would this look, say when we stop parsing region out of the auth token. 

I see two options:

1. `EnvironmentTokenProvider(string authTokenEnvName, string authTokenRegionName)`
2. Or we just do this internally by saying momento always needs you to use certain environment variables - `MOMENTO_AUTH_TOKEN` and `MOMENTO_ENDPOINT`.

Both approaches have pros and cons, but seems like we are shooting for approach #1 here?

Reply:
I think it can't just be region name though, it needs to be endpoint, and that's not something that we've really shipped yet.

IMO we would have a separate Environment provider for different types of tokens?

Reply:
@gautamomento My plan was to have different provider implementations for different purposes. As long as they implement `ICredentialProvider` we'll be fine. Other providers can have all their info passed into the constructor like the `APITokenProvider` example @rlinehan posted or can read the info from files or whatever. The specific provider in this PR is here to support the example scripts and tests, which all read a JWT token from an environment variable.